### DISCUSSION TITLE
Callback on Modal Events (onOpen, onClose)

### DISCUSSION THREAD

**Discussion by owieth** [rainbow-me/rainbowkit#624] [Ideas]
Hi there 👋
First of all thanks for providing this amazing library guys 🥳 

We've found a specific Usecase, where we would like to listen to Events of the Walletmodal.

For Instance, when the User opens the Modal and then closes it again (ESC Key or X-Icon), we want to fire a function so set a specific variable.

We already looked around in the docs of rainbow & wagmi but couldn't find any proper solution for this case 😞 

What do you think guys?

**Comment by basememara** [comment]
I needed this too, to reset the UI to let the user start the flow again.

**Comment by DanielSinclair** [comment]
Open and closed state is now available in the modal hooks. You can track these and fire a callback when they change:
```ts
const { connectModalOpen } = useConnectModal();
const { accountModalOpen } = useAccountModal();
const { chainModalOpen } = useChainModal();
```
